src/button/Button.tsx

```diff
@@ -6,115 +6,115 @@ import Spinner from "../spinner/Spinner";
 // SCSS import is moved here to be able to override spinner styles without nesting
 import "./_button.scss";
 
-export interface ButtonProps {
-  testid?: string;
+export type ButtonProps = Omit<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  "disabled"
+> & {
   children: React.ReactNode;
+  testid?: string;
   customSpinner?: React.ReactNode;
+  isDisabled?: boolean;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  shouldFocus?: boolean;
   onClick?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseOver?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseDown?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseUp?: React.ReactEventHandler<HTMLButtonElement>;
   onFocus?: React.ReactEventHandler<HTMLButtonElement>;
   onBlur?: React.ReactEventHandler<HTMLButtonElement>;
```

Comment:
I think we can also remove those props since it's extended from `React.ButtonHTMLAttributes<HTMLButtonElement>`

```suggestion
```